### ISSUE TITLE
[BUG] Error: Digests do not match

### ISSUE THREAD

**Issue by kryaksy** [microsoft/playwright#14504]
**Context:**
- Playwright Version: 1.21.1
- Operating System: Mac
- Node.js version: 18.2.0
- Browser: Chromium

**Describe the bug**

I've got this error in one of my test case when running multiple test suites with multiple workers. 
"Error: Digests do not match"

What could be the problem here? I'm using playwright for couple of months and this is the first time encountering this issue after setting up some custom fixtures. 

**Some Observations**
- It isn't related with the multiple workers. I've also tried with 1 worker and still got the error. 
- It is however connected with worker count somehow since the error is thrown in different suites with changing worker count.
- It is most probably related to custom fixtures.
- Error is thrown when starting the first case of the test suite and all other cases in the suite is skipped. 
- Also tried in 1.22.2 and got the same error

**Comment by kryaksy** [comment]
@dgozman I didn't think I can make an isolated reproduction but here is:
https://github.com/kryaksy/playwright-digest-do-not-match-repro
It also contains the report from the latest reproduction in case you couldn't reproduce it in your environment. I hope it works though.

**Comment by dgozman** [comment]
@kryaksy The problem here is [this `test.use()` call](https://github.com/kryaksy/playwright-digest-do-not-match-repro/blob/master/src/fixtures/index.ts#L45). Because it is in a shared fixtures file, it is only executed during the first `require()` of that file (`dummy.spec.ts`), so the use overrides apply to the first test file that required the fixtures, but not the others. Unfortunately, that leads to some internal inconsistency in the test runner, which I will look into.

For now, I'd suggest to move `test.use()` into `test.extend()`:
```ts
export const myTest = base.extend<Fixtures>({
  // add fixtures here
  // ....

  // add use options here
  viewport: { width: 1920, height: 1080 },
  trace: { mode: "on", snapshots: true },
  video: { mode: "on", size: { width: 1920, height: 1080 } },
});
```

**Comment by dgozman** [comment]
For the reference, we decided not to proceed with a risky fix in #14598. I'll merge a PR with an improved error message as a mitigation strategy.
